### PR TITLE
Bugfix. Useless auto-fix in AVOID_NULL_CHECKS rule

### LINKED ISSUE #840: Useless auto-fix in AVOID_NULL_CHECKS rule
## Describe the bug
```kotlin
val x = if (y != null) f(y) else null
```
is autofixed into
```kotlin
val x = y?.let {
    f(y)
}
    ?: run {
        null
    }
```
but elvis is redundant here, because `y?.let()` will return `null` itself, if `y == null`.

## Environment information
* diktat version: 0.5.0

### PR DESCRIPTION
**Pull request by aktsay6**
### What's done:
  * Fixed bug
  * Added test

This pull request closes #840

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt [hunk 1]
```diff
@@ -104,7 +104,7 @@ class NullChecksRule(configRules: List<RulesConfig>) : DiktatRule(
                 """.trimMargin()
             }
         } else {
-            if (elseCodeLines.isNullOrEmpty()) {
+            if (elseCodeLines.isNullOrEmpty() || elseCodeLines.first() == "null") {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt:107]: What about weird constructions like
```kotlin
else {
    null
    value
}
```
?

**Review by petertrr — CHANGES_REQUESTED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt (hunk 1 above)
Comment (aktsay6) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt]: Fixed

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt [hunk 2]
```diff
@@ -104,7 +104,7 @@ class NullChecksRule(configRules: List<RulesConfig>) : DiktatRule(
                 """.trimMargin()
             }
         } else {
-            if (elseCodeLines.isNullOrEmpty() || elseCodeLines.first() == "null") {
+            if (elseCodeLines.isNullOrEmpty() || (elseCodeLines.first() == "null" && elseCodeLines.size == 1)) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt:107]:
```suggestion
            if (elseCodeLines.isNullOrEmpty() || (elseCodeLines.singleOrNull() == "null")) {
```
:facepalm:

Comment (aktsay6) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt]: Fixed :D

**Review by petertrr — COMMENTED** [review]
We can add the same logic for the opposite situation as well:
```kotlin
fun foo(y: Int?) {
    val x = if (y == null) {
        null
    } else {
        y + 2
    }
}
```
we shouldn't have useless `run { null }` block

**Review by petertrr — APPROVED** [review]

**Pull request merged**